os/arch/arm/Kconfig

```diff
@@ -271,6 +271,28 @@ config ARCH_DABORTSTACK
 		will be used during data abort handling. Recommended data abort stack
 		size is 1K.
 
+config MPU_STACK_OVERFLOW_PROTECTION
+	bool "MPU Stack Overflow Protection"
+	depends on BUILD_PROTECTED
+	depends on ARM_MPU
+	default n
+	---help---
+		Enables stack overflow protection.
+		It protects each task's stack overflow outside its max allocated stack memory.
+		This is applicable to user mode tasks created in protected build.
+
+if MPU_STACK_OVERFLOW_PROTECTION
+
+config MPU_STACK_GUARD_SIZE
+	int "Stack Guard size configuration range(32-1024)"
+	default 32
+	range 32 1024
+	---help---
+		Set size for the stack guard. It must be set with 32 byte aligned values.
```

Comment:
To check `32 bytes aligned value`, we should add the condition in mkconfig.h

Reply:
added checking condition